src/assembler/parser/error.ts

```diff
@@ -1,18 +1,10 @@
-import { VirtualBoardError, VirtualBoardErrorType } from 'types/error'
+import { AssemblerError } from 'assembler/error'
 import { ITextCursor } from './text'
 
-export class ParseError extends Error {
+export class ParseError extends AssemblerError {
   public cursor: ITextCursor
-  constructor(cursor: ITextCursor, message: string) {
-    super(`${message} at Line ${cursor.line + 1}, Position ${cursor.position}`)
+  constructor(message: string, cursor: ITextCursor) {
+    super(message, cursor.line + 1, cursor.position)
     this.cursor = cursor
   }
```

Comment:
I would leave it there since now all errors are placed in the folder of their respective responsibility. Assembler is for everything inside assembler and InstructionError for everything inside instructions. To keep this theme up I decided to leave the error inside the parser folder because it is just used there.